### ISSUE TITLE
void do_nms_sort_v3(detection *dets, int total, int classes, float thresh)

### ISSUE THREAD

**Issue by XP-CHRC** [AlexeyAB/darknet#648]
@AlexeyAB 
void do_nms_sort_v3(detection *dets, int total, int classes, float thresh)
{
	int i, j, k;
	k = total - 1;
	for (i = 0; i <= k; ++i) {
		if (dets[i].objectness == 0) {
			detection swap = dets[i];
			dets[i] = dets[k];
			dets[k] = swap;
			--k;
			--i;
		}
    } 
dose this step works only when thresh=0 ? Because when doing:
detection *dets = get_network_boxes(&net, im.w, im.h, thresh, hier_thresh, 0, 1, &nboxes, letterbox);
only dets.objectness > thresh were saved.
and after  do_nms_sort_v3() , does there is only one element bigger than 0 in dets[i].prob ? (in other words: one box just detects one objection as the paper said?)
Thank you!

**Comment by AlexeyAB** [comment]
> dose this step works only when thresh=0 ?

* for Yolo v3 yes, there are only with T0 > thresh: https://github.com/AlexeyAB/darknet/blob/9f2dbfa7df981d41c070261c4b9a8a875b290203/src/yolo_layer.c#L322-L324

* for Yolo v2 no, there are all final activations include T0 = 0: https://github.com/AlexeyAB/darknet/blob/9f2dbfa7df981d41c070261c4b9a8a875b290203/src/network.c#L550-L552

----

**Each box can predict many objects in Yolo v3** - this is multilabel approach when used logistic instead of softmax): https://pjreddie.com/media/files/papers/YOLOv3.pdf

> 2.2. Class Prediction
> **Each box predicts the classes the bounding box may contain
> using multilabel classification.** We do not use a softmax
> as we have found it is unnecessary for good performance,
> instead we simply use independent logistic classifiers. During
> training we use binary cross-entropy loss for the class
> predictions.
> This formulation helps when we move to more complex
> domains like the Open Images Dataset [7]. In this dataset
> there are many overlapping labels (i.e. Woman and Person).
> Using a softmax imposes the assumption that each box has
> exactly one class which is often not the case. **A multilabel
> approach better models the data.**
>
